src/domain/ecoverse/ecoverse.service.ts

```diff
@@ -212,6 +216,28 @@ export class EcoverseService {
     return group;
   }
 
+  async createChallenge(challengeData: ChallengeInput): Promise<IChallenge> {
```

Comment:
createChallenge should be in ChallengeService

Reply:
This would create a circular reference in the modules as the challenge also needs to be added into the list of challenges for the ecoverse.

Reply:
No entity is standalone, but the challenge business logic should be in ChallengeService. Circular modular / service reference can be resolved with forwardRef on module and service level (see latest pushed commits on server-159).